pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -648,7 +646,13 @@ func (s *FlinkStateMachine) handleApplicationDeleting(ctx context.Context, app *
 	return nil
 }
 
-func (s *FlinkStateMachine) getAndUpdateError(ctx context.Context, application *v1alpha1.FlinkApplication, err error) {
+func (s *FlinkStateMachine) compareAndUpdateError(ctx context.Context, application *v1alpha1.FlinkApplication, err error) {
```

Comment:
Hey,

Updating the status object in case of non error can fail here, if there has been an update in the handleLoop.

Reply:
@anandswaminathan I've updated the logic now to minimize the number of updateK8Object call. Thanks for the suggestion!